api/src/main/java/ai/djl/Device.java

```diff
@@ -53,12 +53,22 @@ private Device(String deviceType, int deviceId) {
      *
      * @param deviceType the device type, typically CPU or GPU
      * @param deviceId the deviceId on the hardware.
-     * @return a {@code Device} instance
+     * @return a {@code Device} instance, null if not available
      */
     public static Device of(String deviceType, int deviceId) {
         if (Type.CPU.equals(deviceType)) {
             return CPU;
         }
+        if (Type.GPU.equals(deviceType)) {
+            if (getGpuCount() == 0) {
+                return null;
+            }
+            try {
+                CudaUtils.getComputeCapability(deviceId);
```

Comment:
I don't think this is necessary.
Do you see any case that getGpuCount() success but CudaUtils.getComputeCapability(deviceId) failure?

Reply:
I think if deviceId is invalid, say -1 or a value greater than the number of gpu then it could fail.